helm/private/helm_import_authn.bzl

```diff
@@ -11,8 +14,151 @@ _WWW_AUTH = {
     },
 }
 
+def _file_exists(repository_ctx, path):
+    result = repository_ctx.execute(["stat", path])
+    return result.return_code == 0
+
+def _strip_host(url):
+    # TODO: a principled way of doing this
+    return url.replace("http://", "").replace("https://", "").replace("/v1/", "")
+
+# Path of the auth file is determined by the order described here;
+# https://github.com/google/go-containerregistry/tree/main/pkg/authn#tldr-for-consumers-of-this-package
+def _get_auth_file_path(repository_ctx):
+    HOME = repo_utils.get_env_var(repository_ctx, "HOME", "ERR_NO_HOME_SET")
+
+    # this is the standard path where registry credentials are stored
+    # https://docs.docker.com/engine/reference/commandline/cli/#configuration-files
+    DOCKER_CONFIG = "{}/.docker".format(HOME)
+
+    # set DOCKER_CONFIG to $DOCKER_CONFIG env if present
+    if "DOCKER_CONFIG" in repository_ctx.os.environ:
+        DOCKER_CONFIG = repository_ctx.os.environ["DOCKER_CONFIG"]
+
+    config_path = "{}/config.json".format(DOCKER_CONFIG)
+
+    if _file_exists(repository_ctx, config_path):
+        return config_path
+
+    # https://docs.podman.io/en/latest/markdown/podman-login.1.html#authfile-path
+    XDG_RUNTIME_DIR = "{}/.config".format(HOME)
+
+    # set XDG_RUNTIME_DIR to $XDG_RUNTIME_DIR env if present
+    if "XDG_RUNTIME_DIR" in repository_ctx.os.environ:
+        XDG_RUNTIME_DIR = repository_ctx.os.environ["XDG_RUNTIME_DIR"]
+
+    config_path = "{}/containers/auth.json".format(XDG_RUNTIME_DIR)
+
+    # podman support overriding the standard path for the auth file via this special environment variable.
+    # https://docs.podman.io/en/latest/markdown/podman-login.1.html#authfile-path
+    if "REGISTRY_AUTH_FILE" in repository_ctx.os.environ:
+        config_path = repository_ctx.os.environ["REGISTRY_AUTH_FILE"]
+
+    if _file_exists(repository_ctx, config_path):
+        return config_path
+
+    return None
+
+def _fetch_auth_via_creds_helper(repository_ctx, raw_host, helper_name, allow_fail = False):
+    if repository_ctx.os.name.startswith("windows"):
+        executable = "{}.bat".format(helper_name)
+        repository_ctx.file(
+            executable,
+            content = """\
+@echo off
+echo %1 | docker-credential-{} get """.format(helper_name),
```

Comment:
Can you run "which" to find the credential binary and render that into the script? I have found it helpful to see an explicit message about things not being located if PATH is messed up for some reason.